crates/zoon/src/element/button.rs

```diff
@@ -8,6 +8,60 @@ use std::marker::PhantomData;
 
 make_flags!(Label, OnPress);
 
+/// Html Button for your web page.
+/// You can create a new button by using its constructor and chain styling.
+///
+/// # Examples
+/// ```no_run
+/// use zoon::*;
+/// let button = Button::new().s(Align::center()).s(Padding::all(5));
+/// ```
+/// You can also create your button with specific events as well and update the styling the way you need by using [signals](https://crates.io/crates/futures-signals)
+///
+/// 1 - Create signal for local state management.
+///
+/// 2 - The variable **`hovered`** gets updated when the user hovers the button
+///
+/// 3 - The variable **`hovered_signal`** is the notification send by signal
+/// that will actually determine the background color for the button.
+///
+/// # Example
+/// ```no_run
+/// use zoon::{named_color::*, *};
+/// let (hovered, hovered_signal) = Mutable::new_and_signal(false);
+///
+/// let button = Button::new().s(Background::new()
+///     .on_hovered_change(move |is_hovered| hovered.set_neq(is_hovered))
+///     .color_signal(hovered_signal.map_bool_signal(|| GREEN_7, || GREEN_8)));
+/// ```
+/// It is also possible to with local state and make quick updates.
+///
+/// # Example
+/// ```no_run
+/// use zoon::{*, named_color::*};
+/// use std::{rc::Rc,};
+/// use std::borrow::Cow;
+///     
+/// let click_count = Mutable::new(0);
+///
+/// // Create a title that gets updated when counting changes.
+/// let title = click_count.signal().map(|count| {
+///         if count == 0 {
+///             return Cow::from("Click me!");
+///         }
+///         Cow::from(format!("Clicked {}x", count))
+///     });
+///
+/// // Create a raw that contains the button with the label changing and the press method that increment.
+/// let raw_with_a_button = Row::new().item(
+///       Button::new()
+///       .label_signal(title)
+///       .on_press(move || click_count.update(|count| count + 1)),
+///     );
+/// ```
+/// Many more options are aailable on the button.
```

Comment:
Typo in `available`. 
I recommend to enable VS Code (or another IDE) extensions for spell checking.

Reply:
The spell checking is On by default on jetbrain but for some reason since yesterday it stopped working on block comment & Doc after I changes colors to see them. I will figure out why :wink:

Reply:
I solved it, just updated Jetbrain and now it works as normal. I think something was wrong with the state of the IDE-

Reply:
![errors_for_typo](https://user-images.githubusercontent.com/30114057/153708082-0af83fc9-b574-4044-aff2-d2ee27c5e80c.png)